pkg/api/types.go

```diff
@@ -292,6 +300,16 @@ func (a KubernetesAddon) GetAddonContainersIndexByName(containerName string) int
 	return -1
 }
 
+// GetAddonPoolsIndexByName returns the KubernetesAddon pools index with the name `poolName`
+func (a KubernetesAddon) GetAddonPoolsIndexByName(poolName string) int {
```

Comment:
nit: I don't think `pools` needs the `s` here

```suggestion
func (a KubernetesAddon) GetAddonPoolIndexByName(poolName string) int {
```